src/sparse_merkle_tree/smt.rs

```diff
@@ -1391,7 +1394,7 @@ impl SparseMerkleTree {
             if !sorted_queries.is_empty() && query.is_sibling_of(&sorted_queries[0]) {
                 let sibling = sorted_queries.pop_front().unwrap();
                 sibling_hash = Some(sibling.hash);
-            } else if !query.binary_bitmap[0] {
+            } else if !query.binary_bitmap[0] || sibling_hashes.len() == next_sibling_hash {
```

Comment:
In my understanding you are checking by hand if you have consumed all the sibling hashes, because you use the index `next_sibling_hash` rather than popping them (probably for efficiency reasons). 

I still don't understand why you need to do that. It looks to me that you would pass the check even if `query.binary_bitmap[0] == 1` if you have consumed all the sibling hashes. Doesn't this mean that one can append 1 bits to the end of the binary bitmap to enter this branch? I guess the resulting root would be wrong, but I don't see why we need this.

Reply:
I think it's true to say `query.binary_bitmap[0] == 1 if you have consumed all the sibling hash` it will use empty hash, but if it used up all the sibling hash and if `binary_bitmap[0] == 1`, it doesn't have sibling hash to use, so it should be simply error?

Reply:
yes, I think so